util/ipa_ldap.c

```diff
@@ -24,6 +24,35 @@
 
 #include "ipa_ldap.h"
 
+
+/** Print LDAP error message to stderr
+ *
+ */
+void ipa_ldap_error(LDAP *ld, int errnum, char *msg)
+{
+    const char *errstring;
+    char *diagnostic = NULL;
+    int ret = 0;
+    int has_diagnostic = 0;
+
+    /* print custom message msg first. All msg strings have trailing newline
+     */
+    fprintf(stderr, msg);
+
+    /* Get human readable error string and diagnostic message */
+    errstring = ldap_err2string(errnum);
+    ret = ldap_get_option(ld, LDAP_OPT_DIAGNOSTIC_MESSAGE, (void*)&diagnostic);
+    has_diagnostic = ((ret == LDAP_SUCCESS) && diagnostic && *diagnostic);
+
+    if (errstring && has_diagnostic) {
+        fprintf(stderr, "%s: %s\n", errstring, diagnostic);
+    } else if (errstring) {
+        fprintf(stderr, "%s\n", errstring);
+    } else if (has_diagnostic) {
+        fprintf(stderr, "%i: %s", errnum, diagnostic);
+    }
```

Comment:
Can you add a doc-string that msg needs to be new-line terminated otherwise the output will be smashed together?

Should the corner case of errstring=NULL and has_diagnostic=0 be handled?

Reply:
Arguably there is nothing to print in that case. Perhaps I wasn't quite awake yet :/ The docstring might be nice though.

Reply:
I have added a doc string to the new function.